llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -1930,27 +1858,27 @@ static unsigned getWWMRegSpillRestoreOpcode(unsigned Size,
   if (Size != 4)
     llvm_unreachable("unknown wwm register spill size");
 
-  if (IsVectorSuperClass)
+  if (IsVectorSuperClass) // TODO: Always use this if there are AGPRs
```

Comment:
This can be enabled. Did you find any problem while spilling AV_WWM regclasses now?
It can also be done in a separate patch.

Reply:
Haven't tried, it's hard enough to test the base case. WWM can be left for later